tasks/eva_2130/remove_duplicates_in_study.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+from argparse import ArgumentParser
+import pymongo
+from urllib.parse import quote_plus
+
+
+def find_duplicates_and_remove_them(mongo_user, mongo_password, mongo_host, mongo_database, assembly_accession,
+                                    contig_list, study_list, dry_run):
+    duplicates_to_remove_commands = []
+    with get_mongo_connection_handle_url(
+            username=mongo_user,
+            password=mongo_password,
+            host=mongo_host
+    ) as accessioning_mongo_handle:
+        sve_collection = accessioning_mongo_handle[mongo_database]["submittedVariantEntity"]
+        cursor = sve_collection.find({
+            "seq": assembly_accession, "contig": {"$in": contig_list}, "study": {"$in": study_list}
+        })
+        for record in cursor:
+            variants_with_accessions = list(accessioning_mongo_handle[mongo_database]["submittedVariantEntity"].find({'accession': record['accession']}))
+            if len(variants_with_accessions) != 1:
+                print('Found %s duplicates for accession %s' % (len(variants_with_accessions), record['accession']) )
+            for variant in variants_with_accessions:
+                if variant['contig'] != record['contig']:
+                    duplicates_to_remove_commands.append(
+                        pymongo.DeleteOne({'contig': variant['contig'], 'accession': variant['accession']})
+                    )
+        if dry_run:
+            print("Will remove %s variants " % len(duplicates_to_remove_commands))
+        else:
+            sve_collection.bulk_write(requests=duplicates_to_remove_commands, ordered=False)
```

Comment:
I guess this is fine as it's only meant for cleaning up the small (wrong) remnants of the correct_contig_for_study.py script, but if this script is meant for big deletions then I don't know how this will perform.

Reply:
I think for removing 993 variant it should be fine.

Reply:
Just for future information, deleting entries from MongoDB is brutally slow and any issue that prompts us to delete more than a million entries must be inspected more carefully for alternative remediations.